File: app/src/main/java/ru/igla/tfprofiler/utils/StringUtils.java
package ru.igla.tfprofiler.utils;


import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.util.List;
import java.util.Locale;

public final class StringUtils {

    private StringUtils() {
    }

    /***
     * Check string is null or empty
     * @param cs string to check
     * @return true if null or empty, otherwise false
     */
    public static boolean isNullOrEmpty(final @Nullable CharSequence cs) {
        return (cs == null || String.valueOf(cs).trim().length() == 0);
    }

    public static boolean isNullOrEmpty(final @Nullable List<String> list) {
        return list == null || list.isEmpty();
    }

    @NotNull
    public static String getReadableFileSize(long byteSize, boolean includeUnit) {
        if (byteSize < 1024) return "0"; //less than 1 Kb not to show
        final String[] units = new String[]{"B", "KB", "MB", "GB", "TB"};
        int digitGroups = (int) (Math.log10(byteSize) / Math.log10(1024));
        String unit = includeUnit ? " " + units[digitGroups] : "";

        DecimalFormatSymbols otherSymbols = new DecimalFormatSymbols(Locale.US);
        otherSymbols.setDecimalSeparator('.');
        otherSymbols.setGroupingSeparator('.');
        return new DecimalFormat("#,##0.#", otherSymbols).format((int) (byteSize / Math.pow(1024, digitGroups))) + unit;
    }
}

File: app/src/main/java/ru/igla/tfprofiler/utils/StringUtils.kt
package ru.igla.tfprofiler.utils

import java.text.DecimalFormat
import java.text.DecimalFormatSymbols
import java.util.*
import kotlin.math.log10
import kotlin.math.pow

object StringUtils {

    private val units by lazy { arrayOf("B", "KB", "MB", "GB", "TB") }

    /***
     * Check string is null or empty
     * @param cs string to check
     * @return true if null or empty, otherwise false
     */
    @JvmStatic
    fun isNullOrEmpty(cs: CharSequence?): Boolean {
        return cs == null || cs.toString().trim().isEmpty()
    }

    @JvmStatic
    fun getReadableFileSize(byteSize: Long, includeUnit: Boolean): String {
        if (byteSize < 1024L) return "0" //less than 1 Kb not to show
        val digitGroups = (log10(byteSize.toDouble()) / log10(1024.0)).toInt()
        val unit = if (includeUnit) " " + units[digitGroups] else ""
        val otherSymbols = DecimalFormatSymbols(Locale.US)
        otherSymbols.decimalSeparator = '.'
        otherSymbols.groupingSeparator = '.'
        val s = byteSize / 1024.0.pow(digitGroups.toDouble())
        return DecimalFormat("#,##0.#", otherSymbols).format(s.toLong()) + unit
    }
}